models/repo.go

```diff
@@ -278,6 +278,14 @@ func (repo *Repository) APIFormat(mode AccessMode) *api.Repository {
 		Push:  mode >= AccessModeWrite,
 		Pull:  mode >= AccessModeRead,
 	}
+	parent := nil;
+	err = GetBaseRepo()
+	if err != nil {
+		log.Error(4, "GetRepositoryByID[%d]: %v", id, err)
+	}
+	if repo.BaseRepo != nil {
```

Comment:
Adding `populateParent bool` to the argument-list and then `if populateParent && repo.BaseRepo != nil {` should fix the potential infinite loop :)